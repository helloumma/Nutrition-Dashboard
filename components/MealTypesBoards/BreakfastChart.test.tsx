import { render } from "@testing-library/react";
import { expect, it } from "vitest";
import { BreakfastChart } from "..";

it("renders correctly", () => {
  const result = render(<BreakfastChart data={[]} />);
  expect(result).toMatchSnapshot();
});
