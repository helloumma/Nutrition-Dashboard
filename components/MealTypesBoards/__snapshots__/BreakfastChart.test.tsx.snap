// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="border border-black p-2 m-6 text-white"
      >
        <div
          class="recharts-wrapper"
          role="region"
          style="position: relative; cursor: default; width: 350px; height: 350px;"
        >
          <svg
            class="recharts-surface"
            cx="50%"
            cy="50%"
            height="350"
            viewBox="0 0 350 350"
            width="350"
          >
            <title />
            <desc />
            <defs>
              <clippath
                id="recharts1-clip"
              >
                <rect
                  height="340"
                  width="340"
                  x="5"
                  y="5"
                />
              </clippath>
            </defs>
          </svg>
          <div
            class="recharts-legend-wrapper"
            style="position: absolute; width: auto; height: auto; right: 5px; top: 175px;"
          >
            <ul
              class="recharts-default-legend"
              style="padding: 0px; margin: 0px; text-align: left;"
            >
              <li
                class="recharts-legend-item legend-item-0"
                style="display: block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <title />
                  <desc />
                  <path
                    class="recharts-symbols"
                    cx="16"
                    cy="16"
                    d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                    fill="#6ee7b7"
                    transform="translate(16, 16)"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(110, 231, 183);"
                >
                  calories
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-1"
                style="display: block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <title />
                  <desc />
                  <path
                    class="recharts-symbols"
                    cx="16"
                    cy="16"
                    d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                    fill="#06b6d4"
                    transform="translate(16, 16)"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(6, 182, 212);"
                >
                  cholesterol
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-2"
                style="display: block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <title />
                  <desc />
                  <path
                    class="recharts-symbols"
                    cx="16"
                    cy="16"
                    d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                    fill="#93c5fd"
                    transform="translate(16, 16)"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(147, 197, 253);"
                >
                  fiber
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-3"
                style="display: block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <title />
                  <desc />
                  <path
                    class="recharts-symbols"
                    cx="16"
                    cy="16"
                    d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                    fill="#3b82f6"
                    transform="translate(16, 16)"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(59, 130, 246);"
                >
                  potassium
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-4"
                style="display: block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <title />
                  <desc />
                  <path
                    class="recharts-symbols"
                    cx="16"
                    cy="16"
                    d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                    fill="#6366f1"
                    transform="translate(16, 16)"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(99, 102, 241);"
                >
                  protein
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-5"
                style="display: block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <title />
                  <desc />
                  <path
                    class="recharts-symbols"
                    cx="16"
                    cy="16"
                    d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                    fill="#f9a8d4"
                    transform="translate(16, 16)"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(249, 168, 212);"
                >
                  saturated fat
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-6"
                style="display: block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <title />
                  <desc />
                  <path
                    class="recharts-symbols"
                    cx="16"
                    cy="16"
                    d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                    fill="#fb7185"
                    transform="translate(16, 16)"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(251, 113, 133);"
                >
                  sodium
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-7"
                style="display: block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <title />
                  <desc />
                  <path
                    class="recharts-symbols"
                    cx="16"
                    cy="16"
                    d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                    fill="#f43f5e"
                    transform="translate(16, 16)"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(244, 63, 94);"
                >
                  sugars
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-8"
                style="display: block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <title />
                  <desc />
                  <path
                    class="recharts-symbols"
                    cx="16"
                    cy="16"
                    d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                    fill="#fb923c"
                    transform="translate(16, 16)"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(251, 146, 60);"
                >
                  carbohydrates
                </span>
              </li>
              <li
                class="recharts-legend-item legend-item-9"
                style="display: block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <title />
                  <desc />
                  <path
                    class="recharts-symbols"
                    cx="16"
                    cy="16"
                    d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                    fill="#fbbf24"
                    transform="translate(16, 16)"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                  style="color: rgb(251, 191, 36);"
                >
                  total fats
                </span>
              </li>
            </ul>
          </div>
          <div
            class="recharts-tooltip-wrapper"
            role="dialog"
            style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
            tabindex="-1"
          >
            <div
              class="recharts-default-tooltip"
              style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
            >
              <p
                class="recharts-tooltip-label"
                style="margin: 0px;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="border border-black p-2 m-6 text-white"
    >
      <div
        class="recharts-wrapper"
        role="region"
        style="position: relative; cursor: default; width: 350px; height: 350px;"
      >
        <svg
          class="recharts-surface"
          cx="50%"
          cy="50%"
          height="350"
          viewBox="0 0 350 350"
          width="350"
        >
          <title />
          <desc />
          <defs>
            <clippath
              id="recharts1-clip"
            >
              <rect
                height="340"
                width="340"
                x="5"
                y="5"
              />
            </clippath>
          </defs>
        </svg>
        <div
          class="recharts-legend-wrapper"
          style="position: absolute; width: auto; height: auto; right: 5px; top: 175px;"
        >
          <ul
            class="recharts-default-legend"
            style="padding: 0px; margin: 0px; text-align: left;"
          >
            <li
              class="recharts-legend-item legend-item-0"
              style="display: block; margin-right: 10px;"
            >
              <svg
                class="recharts-surface"
                height="14"
                style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                viewBox="0 0 32 32"
                width="14"
              >
                <title />
                <desc />
                <path
                  class="recharts-symbols"
                  cx="16"
                  cy="16"
                  d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                  fill="#6ee7b7"
                  transform="translate(16, 16)"
                />
              </svg>
              <span
                class="recharts-legend-item-text"
                style="color: rgb(110, 231, 183);"
              >
                calories
              </span>
            </li>
            <li
              class="recharts-legend-item legend-item-1"
              style="display: block; margin-right: 10px;"
            >
              <svg
                class="recharts-surface"
                height="14"
                style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                viewBox="0 0 32 32"
                width="14"
              >
                <title />
                <desc />
                <path
                  class="recharts-symbols"
                  cx="16"
                  cy="16"
                  d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                  fill="#06b6d4"
                  transform="translate(16, 16)"
                />
              </svg>
              <span
                class="recharts-legend-item-text"
                style="color: rgb(6, 182, 212);"
              >
                cholesterol
              </span>
            </li>
            <li
              class="recharts-legend-item legend-item-2"
              style="display: block; margin-right: 10px;"
            >
              <svg
                class="recharts-surface"
                height="14"
                style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                viewBox="0 0 32 32"
                width="14"
              >
                <title />
                <desc />
                <path
                  class="recharts-symbols"
                  cx="16"
                  cy="16"
                  d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                  fill="#93c5fd"
                  transform="translate(16, 16)"
                />
              </svg>
              <span
                class="recharts-legend-item-text"
                style="color: rgb(147, 197, 253);"
              >
                fiber
              </span>
            </li>
            <li
              class="recharts-legend-item legend-item-3"
              style="display: block; margin-right: 10px;"
            >
              <svg
                class="recharts-surface"
                height="14"
                style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                viewBox="0 0 32 32"
                width="14"
              >
                <title />
                <desc />
                <path
                  class="recharts-symbols"
                  cx="16"
                  cy="16"
                  d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                  fill="#3b82f6"
                  transform="translate(16, 16)"
                />
              </svg>
              <span
                class="recharts-legend-item-text"
                style="color: rgb(59, 130, 246);"
              >
                potassium
              </span>
            </li>
            <li
              class="recharts-legend-item legend-item-4"
              style="display: block; margin-right: 10px;"
            >
              <svg
                class="recharts-surface"
                height="14"
                style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                viewBox="0 0 32 32"
                width="14"
              >
                <title />
                <desc />
                <path
                  class="recharts-symbols"
                  cx="16"
                  cy="16"
                  d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                  fill="#6366f1"
                  transform="translate(16, 16)"
                />
              </svg>
              <span
                class="recharts-legend-item-text"
                style="color: rgb(99, 102, 241);"
              >
                protein
              </span>
            </li>
            <li
              class="recharts-legend-item legend-item-5"
              style="display: block; margin-right: 10px;"
            >
              <svg
                class="recharts-surface"
                height="14"
                style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                viewBox="0 0 32 32"
                width="14"
              >
                <title />
                <desc />
                <path
                  class="recharts-symbols"
                  cx="16"
                  cy="16"
                  d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                  fill="#f9a8d4"
                  transform="translate(16, 16)"
                />
              </svg>
              <span
                class="recharts-legend-item-text"
                style="color: rgb(249, 168, 212);"
              >
                saturated fat
              </span>
            </li>
            <li
              class="recharts-legend-item legend-item-6"
              style="display: block; margin-right: 10px;"
            >
              <svg
                class="recharts-surface"
                height="14"
                style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                viewBox="0 0 32 32"
                width="14"
              >
                <title />
                <desc />
                <path
                  class="recharts-symbols"
                  cx="16"
                  cy="16"
                  d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                  fill="#fb7185"
                  transform="translate(16, 16)"
                />
              </svg>
              <span
                class="recharts-legend-item-text"
                style="color: rgb(251, 113, 133);"
              >
                sodium
              </span>
            </li>
            <li
              class="recharts-legend-item legend-item-7"
              style="display: block; margin-right: 10px;"
            >
              <svg
                class="recharts-surface"
                height="14"
                style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                viewBox="0 0 32 32"
                width="14"
              >
                <title />
                <desc />
                <path
                  class="recharts-symbols"
                  cx="16"
                  cy="16"
                  d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                  fill="#f43f5e"
                  transform="translate(16, 16)"
                />
              </svg>
              <span
                class="recharts-legend-item-text"
                style="color: rgb(244, 63, 94);"
              >
                sugars
              </span>
            </li>
            <li
              class="recharts-legend-item legend-item-8"
              style="display: block; margin-right: 10px;"
            >
              <svg
                class="recharts-surface"
                height="14"
                style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                viewBox="0 0 32 32"
                width="14"
              >
                <title />
                <desc />
                <path
                  class="recharts-symbols"
                  cx="16"
                  cy="16"
                  d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                  fill="#fb923c"
                  transform="translate(16, 16)"
                />
              </svg>
              <span
                class="recharts-legend-item-text"
                style="color: rgb(251, 146, 60);"
              >
                carbohydrates
              </span>
            </li>
            <li
              class="recharts-legend-item legend-item-9"
              style="display: block; margin-right: 10px;"
            >
              <svg
                class="recharts-surface"
                height="14"
                style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                viewBox="0 0 32 32"
                width="14"
              >
                <title />
                <desc />
                <path
                  class="recharts-symbols"
                  cx="16"
                  cy="16"
                  d="M16,0A16,16,0,1,1,-16,0A16,16,0,1,1,16,0"
                  fill="#fbbf24"
                  transform="translate(16, 16)"
                />
              </svg>
              <span
                class="recharts-legend-item-text"
                style="color: rgb(251, 191, 36);"
              >
                total fats
              </span>
            </li>
          </ul>
        </div>
        <div
          class="recharts-tooltip-wrapper"
          role="dialog"
          style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px; left: 0px;"
          tabindex="-1"
        >
          <div
            class="recharts-default-tooltip"
            style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
          >
            <p
              class="recharts-tooltip-label"
              style="margin: 0px;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
